app/javascript/components/DownloadButton.js

```diff
@@ -29,47 +25,61 @@ function fetchDownloadConfig() {
     'curl -K cfg.txt; rm cfg.txt'
   );
 
-  return [authCode, timeInterval, downloadCommand];
+  return {
+    authCode: authCode,
+    timeInterval: timeInterval,
+    downloadCommand: downloadCommand
+  };
 }
 
 function DownloadCommandContainer() {
-  
-  const [authCode, timeInterval, downloadCommand] = fetchDownloadConfig();
 
-  const expiresMinutes = Math.floor(timeInterval / 60); // seconds -> minutes
+  const [downloadConfig, setDownloadConfig] = useState({});
 
```

Comment:
if I understand useEffect() correctly, this means we will re-fetch the authcode everytime the button rerenders -- is that correct/desired?

Reply:
It seems reasonable, but it's unclear if it's ideal.  SCP-2121 now notes we should consider this more.